### ISSUE TITLE
`task wait` on failed `release deploy` task should exit with an error code

### ISSUE THREAD

**Issue by reddogaw** [OctopusDeploy/cli#390]
## The bug

Using `octopus release deploy <...args...> | octopus task wait` does not result in an error code if the status of the task is failed

At the moment, it always seems to exit with value 0.

This prevents us from chaining deployments together.

## Command to reproduce

```sh
octopus release deploy <...args...> | octopus task wait
if [ $? -ne 0 ]; then
    echo "The last command failed with exit code $?"
    exit 1
else
    echo "The last command succeeded"
fi
```

## Outcome

<img width="906" alt="image" src="https://github.com/user-attachments/assets/4012acf0-3f64-4e9f-995c-157e81cd5c01">

## Versions

**cli**: 2.1.0

**Octopus Server**: 2024.1.12210

**Comment by jameschensmith** [comment]
Seems to still be an issue with the following versions:

**cli**: ServerTasks-257205
**Octopus Server**: 2023.2.12209

**Comment by APErebus** [comment]
This was fixed in `2.10.0` of the CLI
